articles/quickstart/native/ios-objc/03-user-sessions.md

```diff
@@ -121,34 +118,35 @@ Then, if such a token exists, you need to check whether it's still valid, has ex
 }];
 ```
 
-## Dealing with a non-valid accessToken
+## Deal with a Non-Valid Access Token
 
-How to deal with a non-valid accessToken is up to you. You will normally choose between two scenarios:
-Either you ask users to re-enter their credentials, or you can use `.renew(withRefreshToken: refreshToken)` with a [refresh_token](/refresh-token) to obtain a new valid accessToken again.
+Decide how to deal with a non-valid access token. You can choose between two options:
+* Ask users to re-enter their credentials.
+* Use `.renew(withRefreshToken: refreshToken)` with a [refresh_token](/refresh-token) to obtain a new valid access token.
 
-If you aim for the former scenario, make sure you clear all the keychain stored values by doing:
+If you want to ask your users to re-enter their credentials, clear all the values stored in the keychain:
 
 ```objc
 // HomeViewController.m
 A0SimpleKeychain *keychain = [[A0SimpleKeychain alloc] initWithService:@"Auth0"];
 [keychain clearAll];
 ```
 
-However, in this tutorial you will focus on the latter scenario, where you want to log users in without asking for their credentials again.
+The rest of this tutorial shows you how to use a refresh token to obtain a new access token. 
 
-In this case, you're going to leverage the `refreshToken`. The refresh token is another token string contained within the `Credentials` object that comes upon a successful login, which doesn't expire, and whose main purpose is retrieving a new valid `accessToken`.
+The refresh token is a token string stored in the `Credentials` object after a successful login. The refresh token doesn't expire. 
 
 ::: note
-It's recommended that you read and understand the [refresh token documentation](/refresh-token) before proceeding. You should keep in mind, for example, that even though the refresh token cannot expire, it can be revoked.
+Even though the refresh token cannot expire, it can be revoked. For more information, read the [refresh token documentation](/refresh-token) before you proceed with this tutorial.
 :::
 
-### Store the refreshToken
+### Store the refresh token
 
-::: note
-The `refreshToken` can be `nil` if `offline_access` is not sent in the `scope` parameter during authentication.
+::: note 
+If you do not send `offline_access` as a scope during authentication, the refresh token can be `nil`.
```

Comment:
Does this note mean: "it's ok for the token to be `nil`" or "maybe it will be `nil`"?

Reply:
If there is no `offline_access` scope the refreshToken will be nil